File: python/get_article_responses.py
#-- title: Get Article's Responses
#-- description: Code to get Responses/Comments on a Medium Article.
#-- tags: python, medium_api, medium_api_py


# Import libraries
import os
from medium_api import Medium

#%%
# Get RAPIDAPI_KEY from the environment
api_key = os.getenv('RAPIDAPI_KEY')

#%%
# Create a `Medium` Object
medium = Medium(api_key)

#%%
# Get the "Article" object and fetch responses
article = medium.article(article_id="67fa62fc1971")

article.fetch_responses()

#%%
# Iterate over article's responses and print their content
for response in article.responses:
    print(response.content)
    print('\n\n---\n\n')

File: javascript/getArticleResponses.js
// getArticleResponses.js - Fetches the responses for a specific article

import Medium from 'medium-api-js';
import dotenv from 'dotenv';

dotenv.config();

const medium = new Medium(process.env.RAPIDAPI_KEY);

// Specify the article ID for which responses need to be fetched
const articleId = '67fa62fc1971'; 

// Fetch the responses of the article and log the response
medium.getArticleResponses(articleId).then(data => {
    console.log('Article Responses:', data);
});
